commit 4e8507ba774f1fe5cd0c26a7cafc09afb8a6ba8e
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Sun May 6 19:31:47 2018 +0100

    drm/i915: Don't request a bug report for unsafe module parameters
    
    Unsafe module parameters are just that, unsafe. If the user is foolish
    enough to try them and the kernel breaks, they get to keep both pieces.
    Don't ask them to file a bug report if they broke it themselves.
    
    References: https://bugs.freedesktop.org/show_bug.cgi?id=106423
    Fixes: d15d7538c6d2 ("drm/i915: Tune down init error message due to failure injection")
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
    Cc: Imre Deak <imre.deak@intel.com>
    Cc: Jani Nikula <jani.nikula@linux.intel.com>
    Cc: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
    Cc: Rodrigo Vivi <rodrigo.vivi@intel.com>
    Acked-by: Jani Nikula <jani.nikula@intel.com>
    Reviewed-by: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20180506183147.2690-1-chris@chris-wilson.co.uk

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 8c2986849236..2a96d082addf 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -101,7 +101,13 @@ __i915_printk(struct drm_i915_private *dev_priv, const char *level,
 		   __builtin_return_address(0), &vaf);
 
 	if (is_error && !shown_bug_once) {
-		dev_notice(kdev, "%s", FDO_BUG_MSG);
+		/*
+		 * Ask the user to file a bug report for the error, except
+		 * if they may have caused the bug by fiddling with unsafe
+		 * module parameters.
+		 */
+		if (!test_taint(TAINT_USER))
+			dev_notice(kdev, "%s", FDO_BUG_MSG);
 		shown_bug_once = true;
 	}